### ISSUE TITLE
feature: Allow override of i18n messages disabling messagebundle downloads

### ISSUE THREAD

**Issue by codefactor** [SAP/ui5-webcomponents#3071]
**Is your feature request related to a problem? Please describe.**
SuccessFactors has several issues with the messagebundles that come out of the box for certain web components:
1. SuccessFactors supports a different set of languages than those which come out of box, we even have certain "DEBUG" languages which output specialized message values so that a tester can check if any strings on the page show up that do not match the expected obvious debug messages.
1. SuccessFactors supports custom tenant-specific message translations, in cases where a tenant is not satisfied with our own translations
1. When using the webcomponents, typically only a few messages are actually needed, yet the entire messagebundle is downloaded to the browser

Once cd7a494 is merged, we should get a fix where at least the `DELETE` key can be translated -- but there are 2 languages in SuccessFactors that continue to get English for this one message, `bs_ID` (Bahasa Indonesia) and `cy_GB` (Welsh). Also, this one message will not support the debug languages either, but that's a low priority.

There is a feature we can use to override the messages:
https://sap.github.io/ui5-webcomponents/playground/docs/i18n/

But right now we are using version 0.29.X which does not have this feature down-ported. In one of our pages, all of the messages on the screen are coming from our own i18n framework, all except for one -- the `DELETE` key that gets put as the tooltip text for the `UploadCollectionItem`. So it seems like a heavy-weight solution at this point to customize the entire messagebundle for one single tooltip text. Perhaps we can do that in the next release.

**Describe the solution you'd like**

One idea would be to use data attributes:
```html
<ui5-upload-collection data-i18n-DELETE="Delete" mode="Delete">
  <ui5-upload-collection-item />
</ui5-upload-collection>
```
There is a PR for a POC here which involves ideas of adding an `i18nData` object to the `ListItem.prototype` which defines a `DELETE` key; however, at the very least we would need the `i18nData` to be a `Promise`: 
https://github.com/SAP/ui5-webcomponents/pull/3066

**Describe alternatives you've considered**

Perhaps we can think about a lighter-weight solution that can be ported to 0.29.x; however, it is not clear what that light weight solution might be, outside of a hack fix to be able to override the `get deleteText()` function on the `ListItem.prototype`

**Additional context**

**Comment by vladitasev** [comment]
@codefactor I'll try some things today and get back to you. I've also replied to your comments in the PR

**Comment by vladitasev** [comment]
Closing old issues. Please re-open, if still relevant and we'll look into it asap.
